### ISSUE TITLE
How install jdtls

### ISSUE THREAD

**Issue by marcinkk** [ChristianChiarulli/LunarVim#312]
hello i need help configuring vim java development.  i have manjaro linux distributions, how to install jdtls.  describe the installations step by step.

**Comment by ChristianChiarulli** [comment]
It's a nightmare, goodluck

**Comment by alourie** [comment]
I started looking into that, went back to intellij community edition. Way less headache, at least for now.

**Comment by ChristianChiarulli** [comment]
I have it working but I don't blame you at all for this @alourie it really will not be worth it for 99% of people currently. I hope it becomes easier in the future.

**Comment by alourie** [comment]
I agree. Should probably just mark this as "feature request" or "some day"

**Comment by zupatisc** [comment]
Since you are on manjaro you can install jdtls through the AUR, you will need to copy the config it provides though since it tries to write to the folder.

**Comment by marcinkk** [comment]
> Since you are on manjaro you can install jdtls through the AUR, you will need to copy the config it provides though since it tries to write to the folder.

where to copy ??

**Comment by rebuilt** [comment]
I think he's talking about this config.  https://github.com/neovim/nvim-lspconfig/blob/master/CONFIG.md#jdtls

**Comment by jimcornmell** [comment]
Hi I'm trying to get Java working and I have downloaded compiled and installed the java lsp, from here: https://github.com/eclipse/eclipse.jdt.ls .

![Screenshot_20210505_113207](https://user-images.githubusercontent.com/7923416/117128786-c7dfd700-ad95-11eb-87ba-ee6adc53d5a6.png)

When I open a java file Is get "No Avtive Lsp" then after about 1 minute a message which states "100% Starting Java Language Server".  But galaxy line still states "No Active Lsp".

![Screenshot_20210505_113209](https://user-images.githubusercontent.com/7923416/117128804-cc0bf480-ad95-11eb-9adf-ea0f35785462.png)

LspInstall does not have java or jdtls as an option.

I think I'm close but can't quite get there, I've only added "require('lsp.java-ls')" to the init.lua.

LspInfo gives:
```
Configured servers: graphql, emmet_ls, pyright, dartls, yamlls, kotlin_language_server, sumneko_lua, vimls, html, angularls, cssls, clangd, dockerls, solargraph, svelte, tsserver, terraformls, gopls, efm, vuels, intelephense, bashls, rust_analyzer, texlab, jsonls
Neovim logs at: /home/jim/.cache/ranger/nvim/lsp.log

1 client(s) attached to this buffer: jdt.ls

  Client: jdt.ls (id 1)
  	root:      /home/jim/eclipse-workspace/OpenGlTexturedSphere/app
  	filetypes: 
  	cmd:       java-linux-ls
  
1 active client(s): 

  Client: jdt.ls (id 1)
  	root:      /home/jim/eclipse-workspace/OpenGlTexturedSphere/app
  	filetypes: 
  	cmd:       java-linux-ls

Clients that match the filetype java:
```
![Screenshot_20210505_113428](https://user-images.githubusercontent.com/7923416/117128882-e47c0f00-ad95-11eb-9ced-ed4f640bfd0f.png)

Any suggestions?

**Comment by rebuilt** [comment]
I don't know how to set it up but this config just got added for java as an option for setting it up.  https://github.com/neovim/nvim-lspconfig/pull/875#issuecomment-832977095

**Comment by zupatisc** [comment]
@jimcornmell Could you check your lsp.log for any errors regarding jdtls? It's in the cache directory which can be found by running `:lua print(vim.fn.stdpath('cache'))`

**Comment by jimcornmell** [comment]
I've run :PackerSync this morning and LSP is working for Java, ish, although with a delay of 5 seconds until it loads.  Also  GalaxyLine still says "No Active Lsp".  The log has lots of information, I'll wade through it and see what I get working.

**Comment by jimcornmell** [comment]
So java mostly works.  The commit https://github.com/neovim/nvim-lspconfig/pull/875 fixed most issues.   Here is a screenshot of a simple Java class full of issues.

![Screenshot_20210507_132454](https://user-images.githubusercontent.com/7923416/117449526-044b3880-af38-11eb-9941-7a7ef5969d59.png)

There are some issues however:

1) Lsp does not list java and does not allow an installation, I installed jdt.ls by hand.
   LspInstall does not list java as a language.
2) Galaxy Line stays saying "No Active Lsp".
3) It takes ages to start if you have loads of code on your system, there must be a way to restrict how much code it analysis.
4) Lsp only starts 50% of the time, closing a file and reopening usually causes the Lsp to start.

I've merged my changes into my fork of [LunarVim](https://github.com/jimcornmell/LunarVim).    Here are my [full instructions](https://github.com/jimcornmell/LunarVim#java-lsp-install)
